clients/src/main/java/org/apache/kafka/common/network/SslChannelBuilder.java

```diff
@@ -194,7 +202,16 @@ public void close() throws IOException {
          */
         @Override
         public boolean complete() {
+            if (sessionBeginTimeMs == null)
+                sessionBeginTimeMs = time.milliseconds();
```

Comment:
Why are we tracking time here?

Reply:
👍 We don't need this.  I added it in order to support the `long sessionBeginTimeMs()` method I had added to `Authenticator`, but now that the percentage calculation is moved into `SaslClientAuthenticator` we don't need the `sessionBeginTimeMs()` method on `Authenticator` anymore.   This file is now untouched.

Reply:
Again, oops, the file is not untouched because it still needs to send in a `Supplier<Authenticator>` rather than `Authenticator`  when it creates a `KafkaChannel` -- but the reference to `Time` is now gone.